_docs/info/migrate_entire_pool.md

```diff
@@ -0,0 +1,262 @@
+
+``` terminal
+$ zfs get sharenfs tank1/nfs
+NAME       PROPERTY  VALUE                                SOURCE
+tank1/nfs  sharenfs  rw=@192.168.2.92,root=@192.168.2.92  local
+```
+
+The current contents on tank1 which we want to migrate.
+
+``` terminal
+$ zfs list -r tank1
+NAME                  USED  AVAIL  REFER  MOUNTPOINT
+tank1                1.58G  13.4G    24K  /tank1
+tank1/nfs             879M  13.4G   879M  /nfs
+tank1/vm-100-disk-0   734M  13.4G   734M  -
+```
+
+When all this is done it is now time to reboot into single user mode.
+
+After reboot into single user mode we start with creating a snapshot of the entire pool we want to migrate.
+
+``` terminal
+$ sudo zfs snapshot -r tank1@replication
+
+$ zfs list -r -t snapshot tank1
+NAME                              USED  AVAIL  REFER  MOUNTPOINT
+tank1@replication                    0      -    24K  -
+tank1/nfs@replication                0      -   879M  -
+tank1/vm-100-disk-0@replication      0      -   734M  -
+```
+
+After making the a snapshot of the entire pool when temporarily send the entire pool contents to another pool. The pool needs to be configured prior to sending the data and it needs to have a size at least equal to the size of the pool we migrate.
+
+**Local operation**
+
+``` terminal
+$ sudo zfs send -R tank1@replication | sudo zfs recv -Fdu tank2
+
+$ zfs list -r tank2
+NAME                  USED  AVAIL  REFER  MOUNTPOINT
+tank2                1.58G  13.4G    24K  /tank2
+tank2/nfs             879M  13.4G   879M  /nfs
+tank2/vm-100-disk-0   734M  13.4G   734M  -
+
+$ zfs list -r -t snapshot tank2
+NAME                              USED  AVAIL  REFER  MOUNTPOINT
+tank2@replication                    0      -    24K  -
+tank2/nfs@replication                0      -   879M  -
+tank2/vm-100-disk-0@replication      0      -   734M  - z
+```
+
+**Remote operation**
+
+``` terminal
+$ sudo zfs send -R tank1@replication | ssh <destination> sudo zfs recv -Fdu tank2
+
+$ ssh <destination> zfs list -r tank2
+NAME                  USED  AVAIL  REFER  MOUNTPOINT
+tank2                1.58G  13.4G    24K  /tank2
+tank2/nfs             879M  13.4G   879M  /nfs
+tank2/vm-100-disk-0   734M  13.4G   734M  -
+
+$ ssh <destination> zfs list -r -t snapshot tank2
+NAME                              USED  AVAIL  REFER  MOUNTPOINT
+tank2@replication                    0      -    24K  -
+tank2/nfs@replication                0      -   879M  -
+tank2/vm-100-disk-0@replication      0      -   734M  - z
+```
+
+After we have insured our destination pool contains the same as our source pool we can safely destroy the replication snapshots on both source and destination pools as well as destroying the old source pool and recreate a new source pool. As stated at the beginning if you want to use another name for the new source pool at extra operation is needed which is shown at the end.
+
+**Local operation**
+
+``` terminal
+$ sudo zfs destroy -r tank1@replication
```

Comment:
you could write `local-host $` instead of `$` to make it clear where the command is to be executed ...

Reply:
Fixed in latest commit.